CMakeLists.txt

```diff
@@ -103,9 +110,15 @@ endif()
 
 # configuring boost
 set(Boost_USE_STATIC_LIBS OFF)
-find_package(Boost 1.81.0 REQUIRED COMPONENTS container thread graph CONFIG)
-include_directories(${Boost_INCLUDE_DIRS})
-message(${Boost_INCLUDE_DIRS})
+
+find_package(Boost 1.81.0 REQUIRED COMPONENTS container thread graph)
+if (Boost_FOUND) 
+    message(STATUS "Found Boost include dirs: ${Boost_INCLUDE_DIRS}")
+    include_directories(${Boost_INCLUDE_DIRS})
+    message(STATUS "Found Boost library dirs: ${Boost_LIBRARY_DIRS}")
```

Comment:
As for "Modern CMake", I encountered a problem when decide to add this log. In 3c34d42 we set default behaviour to searching for BoostConfig.cmake. And if we look at [config](https://github.com/boostorg/boost_install/blob/d77d0e94ba3d78f2079313ecb6e4907a7e8618e9/BoostConfig.cmake#L226) there is a section "compatibility variables" that contains setting of Boost_INCLUDE_DIRS, but not Boost_LIBRARY_DIRS, so log was empty.

Why like this? I found a discussion there one of boost contributors [answered](https://lists.boost.org/Archives/boost/2021/07/251772.php): 
"""
> but Boost_LIBRARY_DIRS is not set anymore - but maybe its just not needed

All of them are not needed at all and not set by the CONFIG files. The
FindBoost module has explicit code to add those for legacy reasons.
In "good" CMake you are supposed to use the targets so all those
variables don't matter for ages now.
"""
So this is about "Modern CMake" approach. CMake forces to use targets, it can also be seen in [include_directories documentation](https://cmake.org/cmake/help/latest/command/include_directories.html). Our CMake script not use this approach and we should fix it at least for such compatibility reasons. 

So as this compatibility was included for legacy reasons I suggest to stay use Boost find module and fix this later.